pkg/services/ngalert/metrics/remote_secondary_forked_alertmanager.go

```diff
@@ -0,0 +1,35 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type RemoteSecondaryForkedAlertmanager struct {
+	SyncsTotal                    *prometheus.CounterVec
+	ConfigurationSyncsFailedTotal *prometheus.CounterVec
+	StateSyncsFailedTotal         *prometheus.CounterVec
+}
+
+func NewRemoteSecondaryForkedAlertmanagerMetrics(r prometheus.Registerer, subsystem string) *RemoteSecondaryForkedAlertmanager {
+	return &RemoteSecondaryForkedAlertmanager{
+		SyncsTotal: promauto.With(r).NewCounterVec(prometheus.CounterOpts{
+			Namespace: Namespace,
+			Subsystem: subsystem,
+			Name:      "remote_secondary_forked_alertmanager_syncs_total",
```

Comment:
Do these metrics apply specific to the remote **secondary** alertmanager? They seem like they'd make sense for any remote alertmanager, even in primary mode.

If so, could we consider dropping `secondary` from all the names?

Also, `forked` feels like an implementation detail. Could it be `remote_alertmanager_syncs_total` and similar?

Reply:
This is only for remote secondary, we won't sync config and state in other modes, but rather redirect writes to the remote Alertmanager. There won't be any time-based convergence.

Reply:
Agree with alex and I would take it one step further and and get rid of any sort of attachment to any of the modes. Metrics just exists as-is and are used by whatever mode we're on. 

As such, everything should be in the remote alertmanager metrics struct.